x-pack/platform/plugins/shared/streams_app/test/scout/ui/tests/data_management/data_retention.spec.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable playwright/max-nested-describe */
+
+import { expect } from '@kbn/scout';
+import { test } from '../../fixtures';
+
+test.describe('Stream data retention', { tag: ['@ess', '@svlOblt'] }, () => {
+  test.beforeAll(async ({ apiServices }) => {
+    await apiServices.streams.enable();
+  });
+
+  test.beforeEach(async ({ browserAuth }) => {
+    await browserAuth.loginAsAdmin();
+  });
+
+  test.afterAll(async ({ apiServices }) => {
+    await apiServices.streams.disable();
+  });
+
+  test.describe('Updating data retention', () => {
```

Comment:
I think we disabled nested `describe` blocks in tests for few reasons:
- Our CI bot can only skip at the _top-level describe_, so nesting causes us to lose unaffected tests when one is flaky.
- Scout runs tests in parallel at the file level when you can do so; nesting limits this and can lead to shared state issues.
- Flat structure keeps tests clear and easier to maintain. It’s easier to debug and selectively run tests with one describe per file.

Playwright recommends flat test hierarchies for these reasons.

Reply:
Btw, if your goal is more granular report, you can use `test.step` decorator for long e2e workflows (probably not the case here)

Reply:
The reason I went for a more granular nesting is about the usage of before/after hooks.
Some tests don't need to load specific data or do some sequential clean-ups, while grouping helped to keep it clean and organized. With a flat hierarchy is clearly possible to achieve it anyway, although I have to mix the tests execution with the setup steps in the tests definition for those that need it, making it more difficult to maintain and update in the long run. Do you have any other recommendation to have setup with the before/after hooks for some groups only?

Reply:
Aha, that's the next spec file :) 
Here I will get back my first point: I encourage against multiple nested describe blocks in favour of multiple spec files. Having too many tests in the same file puts you under the risk of all tests being skipped even if a single block fails. There is no practical need to drop number of spec files, literally the opposite - re-running on failure will be faster and simpler.

Reply:
Ouch you are absolutely right, my bad! I can surely flatten this file!